### PR TITLE
Musl static build

### PR DESCRIPTION
**Pull request by c-nixon**
This adds the ability to build a completely static agent binary using musl.

The binary is still built in docker using the existing build-image command. to configure it set the STATIC env var to 1 when calling make build-image

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,18 +15,31 @@ ENV SCCACHE_BUCKET=${SCCACHE_BUCKET}
...
 COPY . .
 
+RUN env
 # Rebuild the agent
 RUN --mount=type=secret,id=aws,target=/root/.aws/credentials \
     if [ -z "$SCCACHE_BUCKET" ]; then unset RUSTC_WRAPPER; fi; \
-    cargo build --manifest-path bin/Cargo.toml ${FEATURES} --release && strip ./target/release/logdna-agent && \
+    export ${BUILD_ENVS}; cargo build --manifest-path bin/Cargo.toml ${FEATURES} --release --target ${TARGET} && \
+    strip ./target/${TARGET}/release/logdna-agent && \
+    cp ./target/${TARGET}/release/logdna-agent /logdna-agent && \
```
Comment (jorgebay) [inline Dockerfile:41]: oh, good idea to simplify resulting path here.

**Review by jorgebay — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -287,6 +313,10 @@ build-image: ## Build a docker image as specified in the Dockerfile
 		--build-arg SCCACHE_BUCKET=$(SCCACHE_BUCKET) \
 		--build-arg SCCACHE_REGION=$(SCCACHE_REGION)
 
+.PHONY: publish-s3-binary
+publish-s3-binary:
+	aws s3 cp target/$(TARGET)/release/logdna-agent s3://logdna-agent-build-bin/$(TARGET_TAG)/$(TARGET)/logdna-agent
```
Comment (jorgebay) [inline Makefile:318]: Use --acl public-read to allow public reading

File: Makefile [hunk 3]
```diff
@@ -273,11 +296,14 @@ release: ## Create a new release from the current beta and push to github
 .PHONY:build-image
 build-image: ## Build a docker image as specified in the Dockerfile
 	$(DOCKER) build . -t $(REPO):$(BUILD_TAG) \
-		--pull \
+		$(PULL_OPTS) \
```
Comment (jorgebay) [inline Makefile:299]: Curious, why the change to pull newer image?

**Review by jorgebay — APPROVED** [review]
lgtm!

Just a question below.

File: Makefile (hunk 3 above)
Comment (c-nixon) [reply Makefile:299]: This just makes it use the PULL_OPTS option that was already in the makefile, it shouldn't change the behaviour except that it means we can override it by setitng PULL_OPTS=0

Comment (jorgebay) [reply Makefile:299]: makes sense
